lib/manager/npm/post-update/yarn.ts

```diff
@@ -81,21 +82,21 @@ export async function generateLockFile(
     const isYarnDedupeAvailable =
       minYarnVersion && gte(minYarnVersion, '2.2.0');
     const isYarnModeAvailable = minYarnVersion && gte(minYarnVersion, '3.0.0');
-
-    let installYarn = 'npm i -g yarn';
+    const toolConstraint: ToolConstraint = {
+      toolName: 'yarn',
+    };
     if (isYarn1 && minYarnVersion) {
-      installYarn += `@${quote(yarnCompatibility)}`;
+      toolConstraint.constraint = yarnCompatibility;
```

Comment:
The Yarn 1 vs 2/3 works is non-standard and we should double check that this logic still holds. If this `if` is false then we just install latest `yarn`, which is currently v1

Reply:
yes, we still need a global yarn v1

Reply:
Sounds like our logic is still ok here